src/extensions/server/mod.rs

```diff
@@ -176,10 +177,16 @@ impl SubwayServerBuilder {
                                 .map(|(a, b, c)| layer_fn(|s| PrometheusService::new(s, protocol, a, b, c))),
                         );
 
+                    let batch_request_config = match config.max_batch_size {
+                        Some(max_size) => BatchRequestConfig::Limit(max_size),
+                        None => BatchRequestConfig::Unlimited,
```

Comment:
I don't think I want the default to be unlimited. if it is not set, it should be disabled. and we don't really need to support the unlimited case because it shouldn't be used in production anyway.

Reply:
`Disable` means all batch requests are denied. 
Is it your expected behaviour?

```
pub enum BatchRequestConfig {
	/// Batch requests are disabled.
	Disabled,
	/// Each batch request is limited to `len` and any batch request bigger than `len` will not be processed.
	Limit(u32),
	/// The batch request is unlimited.
	Unlimited,
}
```
Because jsonrpsee side implementation is `Unlimited` by default https://github.com/paritytech/jsonrpsee/blob/a4533966b997e83632509ad97eea010fc7c3efc0/server/src/server.rs#L348, so current Subway allows unlimited batch size.

Reply:
ok. I didn't know that. in that case it should be fine. maybe map size 0 to disable? does it make a difference?

Reply:
Setting 0 works but getting different error message.

`Limit(0)` -> `{"jsonrpc":"2.0","error":{"code":-32010,"message":"The batch request was too large","data":"Exceeded max limit of 0"},"id":null}`

`Disabled` -> `{"jsonrpc":"2.0","error":{"code":-32005,"message":"Batched requests are not supported by this server"},"id":null}`

I think defaulting to `Disabled` would be nicer here, but need to make sure to notify existing users about this.

Reply:
or we can make it a non optional type and force user to set a value

Reply:
That's a good idea!